src/OracleService/OracleService.cs

```diff
@@ -0,0 +1,495 @@
+using Akka.Actor;
+using Akka.Util.Internal;
+using Microsoft.Extensions.Configuration;
+using Neo.ConsoleService;
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Network.P2P;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.VM;
+using Neo.Wallets;
+using Neo.Wallets.NEP6;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Timers;
+using static System.IO.Path;
+using NJArray = Newtonsoft.Json.Linq.JArray;
+using NJObject = Newtonsoft.Json.Linq.JObject;
+using NUtility = Neo.Utility;
+
+namespace Neo.Plugins
+{
+    public class OracleService : Plugin, IPersistencePlugin
+    {
+        private NEP6Wallet Wallet;
+        private string[] Nodes;
+        private TimeSpan MaxTaskTimeout;
+        private ConcurrentDictionary<ulong, OracleTask> PendingQueue;
+        private CancellationTokenSource CancelSource;
+        private int Counter;
+        private ConcurrentDictionary<ulong, DateTime> FinishedCache;
+        private ConsoleServiceBase ConsoleBase;
+        private System.Timers.Timer Timer;
+
+        private static readonly object _lock = new object();
+
+        private const int RefreshInterval = 1000 * 60 * 3;
+
+        private static readonly OracleHttpProtocol Https = new OracleHttpProtocol();
+
+        public override string Description => "Built-in oracle plugin";
+
+        public OracleService()
+        {
+            PendingQueue = new ConcurrentDictionary<ulong, OracleTask>();
+            FinishedCache = new ConcurrentDictionary<ulong, DateTime>();
+
+            RpcServerPlugin.RegisterMethods(this);
+
+            ConsoleBase = GetService<ConsoleServiceBase>();
+        }
+
+        protected override void Configure()
+        {
+            var config = GetConfiguration();
+            Wallet = new NEP6Wallet(Combine(PluginsDirectory, nameof(OracleService), config.GetSection("Wallet").Value));
+            Nodes = config.GetSection("Nodes").GetChildren().Select(p => p.Get<string>()).ToArray();
+            MaxTaskTimeout = TimeSpan.FromMilliseconds(double.Parse(config.GetSection("MaxTaskTimeout").Value));
+            OracleHttpProtocol.Timeout = int.Parse(config.GetSection("HttpsTimeout").Value);
+            OracleHttpProtocol.AllowPrivateHost = bool.Parse(config.GetSection("HttpsAllowPrivateHost").Value);
+            OracleHttpProtocol.AllowedContentTypes = config.GetSection("HttpsAllowedContentTypes").GetChildren().Select(p => p.Get<string>()).ToArray();
```

Comment:
I think `AllowPrivateHost` and `AllowedContentTypes` are not for `https` only. They should be global and affects all protocols.

Reply:
Done